File: BaseLibrary/utils/RobotUtils.cs
using System;
using BaseLibrary.battlefield;

namespace BaseLibrary.utils {
    public static class RobotUtils {

        /// <summary>
        /// Get robots vertical speed.
        /// </summary>
        /// <param name="r"></param>
        /// <returns>Robot's vertical speed</returns>
        public static double GetSpeedX(Robot r) {
            return r.Power * r.Motor.MAX_SPEED / 100.0 * Math.Cos(AngleUtils.ToRads(r.AngleDrive));
        }


        /// <summary>
        /// Get robots horizontal speed.
        /// </summary>
        /// <param name="r"></param>
        /// <returns>Robot's horizontal speed</returns>
        public static double GetSpeedY(Robot r) {
            return r.Power * r.Motor.MAX_SPEED / 100.0 * Math.Sin(AngleUtils.ToRads(r.AngleDrive));
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/utils/RobotUtils.java
package BaseLibrary.utils;

import BaseLibrary.battlefield.Robot;

public final class RobotUtils {
	private RobotUtils() {}
	
	public static double getSpeedX(Robot r) {
		return r.Power * r.Motor.MAX_SPEED / 100.0 * Math.cos(AngleUtils.toRads(r.AngleDrive));
	}

	public static double getSpeedY(Robot r) {
		return r.Power * r.Motor.MAX_SPEED / 100.0 * Math.sin(AngleUtils.toRads(r.AngleDrive));
	}
}
